experimental/client-csharp/src/Wandb/Settings.cs

```diff
@@ -0,0 +1,298 @@
+        /// </summary>
+        public string Mode { get; }
+
+        /// <summary>
+        /// Gets or sets the project name under which the run is logged.
+        /// </summary>
+        public string Project { get; set; }
+
+        /// <summary>
+        /// Gets or sets the resume option for the run.
+        /// </summary>
+        public ResumeOption? Resume { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the run has resumed from a previous state.
+        /// </summary>
+        public bool Resumed { get; set; }
+
+        /// <summary>
+        /// Gets the unique identifier for the run.
+        /// </summary>
+        public string RunId { get; }
+
+        /// <summary>
+        /// Gets or sets the amount of time (in seconds) to wait for wandb-core to launch.
+        /// </summary>
+        public float ServiceWait { get; set; }
+
+        /// <summary>
+        /// Gets or sets the run initialization timeout (in seconds).
+        /// </summary>
+        public float InitTimeout { get; set; }
+
+        /// <summary>
+        /// Gets or sets the start date and time of the run.
+        /// </summary>
+        public DateTime StartDatetime { get; set; }
+
+        /// <summary>
+        /// Converts the <see cref="Resume"/> option to its string representation.
+        /// </summary>
+        /// <returns>
+        /// A lowercase string of the resume option, or <c>null</c> if not set.
+        /// </returns>
+        public string ResumeOptionToString()
+        {
+            if (!Resume.HasValue)
+            {
+                return null;
+            }
+            else
+            {
+                return Resume.Value.ToString().ToLowerInvariant();
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Settings"/> class with optional parameters.
+        /// </summary>
+        /// <param name="apiKey">The API key for authentication.</param>
+        /// <param name="baseUrl">The base URL of the wandb server.</param>
+        /// <param name="displayName">The display name of the run.</param>
+        /// <param name="entity">The entity under which the run is logged.</param>
+        /// <param name="mode">The mode in which the run operates.</param>
+        /// <param name="project">The project name.</param>
+        /// <param name="resume">The resume option.</param>
+        /// <param name="runId">The unique identifier for the run.</param>
+        /// <param name="serviceWait">The service wait time in seconds.</param>
+        /// <param name="initTimeout">The initialization timeout in seconds.</param>
+        public Settings(
+            string? apiKey = null,
+            string? baseUrl = null,
+            string? displayName = null,
+            string? entity = null,
+            string? mode = null,
+            string? project = null,
+            ResumeOption? resume = null,
```

Comment:
We should probably provide a default value here of how we want the resume functionality to work.

Without digging too deeply it is not clear what the default behavior of the client is for resuming a run.

Reply:
the user should choose an option, see https://docs.wandb.ai/guides/runs/resuming. if it's not set, we overwrite the run if it exists.
we do have the "auto" option in the python client, but historically it's caused a lot of confusion.